### ISSUE TITLE
Predicates in arrays are not evaluated

### ISSUE THREAD

**Issue by DrRataplan** [FontoXML/fontoxpath#307]
[playground link](https://xpath.playground.fontoxml.com/?mode=1&variables=%7B%7D&xml=%3Croot%3E%0A%3Cfx-instance+id%3D%22countries%22+as-json%3D%22true%22%3E%0A%5B%0A++%7B%22code%22%3A+%22en%22%2C+%22name%22%3A+%22England%22%7D%2C+%0A++%7B%22code%22%3A+%22de%22%2C+%22name%22%3A+%22Germany%22%7D%2C+%0A++%7B%22code%22%3A+%22au%22%2C+%22name%22%3A+%22Australia%22%7D%0A%5D%0A%3C%2Ffx-instance%3E%0A%3Cfx-instance+id%3D%22continents%22+as-json%3D%22true%22%3E%0A%5B%0A++%7B+%22name%22%3A+%22Australia%22%2C+%22countries%22%3A+%5B%22au%22%5D%7D%2C%0A++%7B+%22name%22%3A+%22Europe%22%2C+%22countries%22%3A+%5B%22en%22%2C+%22de%22%5D%7D%0A%5D%0A%3C%2Ffx-instance%3E%0A%3C%2Froot%3E%0A&xpath=declare+namespace+xforms%3D%22i-do-not-remember-the-uri%22%3B%0A%0Adeclare+variable+%24root+%3A%3D+%2F%3B%0A%0Adeclare+function+xforms%3Ainstance%28%24id%29+%7B%0A%09%24root%2F%2Ffx-instance%5B%40id%3D%24id%5D%0A++++%3D%3E+parse-json%28%29%0A%7D%3B%0A%0Adeclare+function+local%3Afilter+%28%24country%29+%7B%0A++%24country%0A%7D%3B%0A%0Adeclare+variable+%24selected-continent+%3A%3D+%22Europe%22%3B%0Adeclare+variable+%24continents+%3A%3D+xforms%3Ainstance%28%22continents%22%29%3B%0Adeclare+variable+%24countries+%3A%3D+xforms%3Ainstance%28%22countries%22%29%3B%0A%0Adeclare+function+local%3Acodes-by-continent%28%24continent%29+%7B%0A%09array%3Afilter%28%24continents%2C+function+%28%24c%29+%7B%0A++++%09%24c%3Fname%3D%24continent%0A++++%7D%29%3F1%3Fcountries%3F*%0A%7D%3B%0A%0A%0Aarray%3Afilter%28%24countries%2C+function+%28%24country%29+%7B%0A++%24country%3Fcode+%3D+local%3Acodes-by-continent%28%24selected-continent%29%0A%7D%29%2C%0A%27----%27%2C%0A%24countries%3F*%5B%3Fcode%3D%28%24continents%3F*%5B%3Fname%3D%24selected-continent%5D%29%3F*%5D%0A%0A%0A&context=)

Basically, the playground should have output the same array twice, however, the filter using the predicates is not working.
